src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs

```diff
@@ -130,6 +130,18 @@ public void AddProject(PackageSpec projectSpec)
             _projects.Add(projectUniqueName, projectSpec);
         }
 
+        /// <summary>
+        /// Add all restorable projects to the restore list.
+        /// This is the behavior for --recursive
+        /// </summary>
+        public void AddRestoreForAllProjects()
```

Comment:
This class should be generic data structure-like container with no logic. While this method suggests content altering based on external criteria tailored to fit specific usage scenario. It ought to be placed in a builder class `MSBuildRestoreUtility`.